Database and DAO added

ShafinMahmudCS opened the pull request:
Can you please review the Database and DAO before it gets merged into the development branch?

ShafinMahmudCS committed:
Database setup for the Merch Store

Why the addition of the Database?
The database would allow for efficient storage, retrieval, and manipulation of data. They will help to ensure that data is accurate, consistent, and easily accessible when it is needed.

Directory: backend\src\main\resources\db.migration

What other things will your change affect?
N/A

Abudy72 committed:
Fixed pathing for Flyway service

ShafinMahmudCS committed:
Database updated and DAO added

ShafinMahmudCS committed:
DAO updated

Abudy72 commented on backend/src/main/java/com/MerchStore/backend/Dao/CartDao.java:
```diff
@@ -0,0 +1,99 @@
+
+    @Override
+    public boolean update(Cart cart) {
+        String statement = "UPDATE cart set (cart_id, user_id) = (?, ?) where cart_id = ?";
```
I dont think there would be a scenario where we reassign an existing cart to someone else.

ShafinMahmudCS replied:
I have changed the update method to throw UnsupportedOperationException.

Abudy72 commented on backend/src/main/java/com/MerchStore/backend/Dao/CartListDao.java:
```diff
@@ -0,0 +1,100 @@
+
+    @Override
+    public boolean update(CartList cart_list) {
+        String statement = "UPDATE cart_list set (cart_id, product_id, product_quantity) = (?, ?, ?) where cart_id = ?";
```
Same thing here, we wont be updating existing cart_ids, they are used to identify each row uniquely.

ShafinMahmudCS replied:
I have changed the update method to just update the product quantity.

Abudy72 commented on backend/src/main/java/com/MerchStore/backend/Model/Order.java:
```diff
@@ -5,15 +5,15 @@
     private final long cartId;
 
-    public Order(long orderId, int quantity, double totalAmount, String orderStatus, long cartId){
+    public Order(long orderId, long quantity, double totalAmount, String orderStatus, long cartId){
```
I dont think someone would order a large amount of any given product, int was just fine!

ShafinMahmudCS replied:
I have changed the data type of quantity to int.

Abudy72 commented on backend/src/main/java/com/MerchStore/backend/Model/Product.java:
```diff
@@ -10,10 +10,10 @@ public class Product {
     private double price;
 
-    private int stock;
+    private long stock;
```
long is an overkill for stock too, i hardly doubt we would have large inventory.

ShafinMahmudCS replied:
I have changed the data type of stock to int.

Abudy72 committed:
adjusted database scripts

ShafinMahmudCS committed:
Changed data type and fixed DAO methods

Changed the data type of stock and quantity from long to int.

Fixed some DAO methods like update and delete.

ShafinMahmudCS committed:
Update method changed

Update method of OrderDao and OrderedItemsDao were changed to throw an exception

jason-obj reviewed (approved):
Nothing catches my eye on the first glance, things look good great work.

LGTM
